### ISSUE TITLE
[$250] Attachment - Wrong video playback speed is highlighted

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42425]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.74.4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing** https://github.com/Expensify/Expensify/issues/383873
## Action Performed:
1. Log in with any account
2. Open any chat or room
3. Upload a video attachment
4. Play the video
5. Click on the "..." button
6. Click on the "Playback speed" button
7. Change the speed
8. Click on the "Playback speed" button again

## Expected Result:
The current playback speed should be highlighted

## Actual Result:
Wrong video playback speed is highlighted. There is no highlight on Android and iOS app

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/a9df940b-70eb-4874-b4d8-2c1fd7ed2442

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bf77b70eb73d8074</li>
        <li>Upwork Job ID: 1793174201872076800</li>
        <li>Last Price Increase: 2024-06-26</li>
<li>Automatic offers: </li>
<ul>
<li>Krishna2323 | Contributor | 102998073</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @abdulrahuman5196</details>

**Comment by lanitochka17** [comment]
@JmillsExpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by JmillsExpensify** [comment]
Opening up to the community.

**Comment by hurram-dev** [comment]
**Proposal**


**Please re-state the problem that we are trying to solve in this issue.**
While playing attachment video, wrong video playback speed is highlighted on the web. There is no highlight on Android and iOS app.

**What is the root cause of that problem?**
The actual cause for this problem is `focus` state is not changing according to selected menu. its value is persisting the initial value that is equal to index of `0.5` speed. So, the highlight of menu item is not working after selecting the menu item

**What changes do you think we should make in order to solve the problem?**
```
 const selectItem = (index: number) => {
        const selectedItem = currentMenuItems[index];
        if (selectedItem?.subMenuItems) {
            setCurrentMenuItems([...selectedItem.subMenuItems]);
            setEnteredSubMenuIndexes([...enteredSubMenuIndexes, index]);
            
            setFocusedIndex(index) //this state needs to be updated after selecting item and it should only work for subMenutItems since focused index is not needed for normal menu item
        } else {
            selectedItemIndex.current = index;
            onItemSelected(selectedItem, index);
        }
    };
```

**What alternative solutions did you explore? (Optional)**

Another solution for this problem would be passing `setFocusedIndex` to 
https://github.com/Expensify/App/blob/e8ae3c5acedf0e6788dc574c7b6f3043ca37092a/src/components/PopoverMenu.tsx#L160

this above function call and update focused index according to selected speed menu item index to get it highlighted

From this [App](https://github.com/Expensify/App/tree/main)/[src](https://github.com/Expensify/App/tree/main/src)/[components](https://github.com/Expensify/App/tree/main/src/components)/[VideoPlayerContexts](https://github.com/Expensify/App/tree/main/src/components/VideoPlayerContexts)
/VideoPopoverMenuContext.tsx file i would change this code as following

```
 items.push({
            icon: Expensicons.Meter,
            text: translate('videoPlayer.playbackSpeed'),
            subMenuItems: CONST.VIDEO_PLAYER.PLAYBACK_SPEEDS.map((speed, index) => ({
                icon: currentPlaybackSpeed === speed ? Expensicons.Checkmark : undefined,
                text: speed.toString(),
                onSelected: (setFocusedIndex //receiving setstate from argument) => {
                    updatePlaybackSpeed(speed);
                    
                    setFocusedIndex(index); //while updating the speed, it also updates menu focus as well and get highlighted
                },
                shouldPutLeftPaddingWhenNoIcon: true,
            })),
        });
```

Let me know, if this proposal seems to work

**Comment by hurram-dev** [comment]
Contributor details
Expensify account email: xurrambeks@gmail.com
Upwork Profile: https://www.upwork.com/freelancers/~0130a0ff1a64915163

**Comment by JmillsExpensify** [comment]
Waiting for proposal reviews. cc @abdulrahuman5196

**Comment by JmillsExpensify** [comment]
Same as above.

**Comment by abdulrahuman5196** [comment]
Hi, Will check on this tomorrow

**Comment by abdulrahuman5196** [comment]
Checking now

**Comment by abdulrahuman5196** [comment]
@lanitochka17 / @JmillsExpensify I don't see this happening from the OP - **Wrong video playback speed is highlighted**.
But I can see this in chrome web - **There is no highlight on Android and iOS app**

Should we make a fix to show the pre-selected playback speed highlighted?


https://github.com/Expensify/App/assets/46707890/3bfd57f9-2c2b-45a4-a2e9-dd25651b540a

**Comment by JmillsExpensify** [comment]
I'm good with that appraoch @abdulrahuman5196

**Comment by abdulrahuman5196** [comment]
Hi, Sorry I was OOO couple of days, Will check the proposals as per the mentioned approach in my morning

**Comment by JmillsExpensify** [comment]
Still waiting on proposal review/next steps.

**Comment by abdulrahuman5196** [comment]
Hi, @JmillsExpensify Sorry for the delay, I have limited availability at the moment. So unassigning myself. Kindly reassign another C+ for review.

**Comment by allroundexperts** [comment]
I can take this on!

**Comment by JmillsExpensify** [comment]
All yours @allroundexperts!

**Comment by allroundexperts** [comment]
Thanks for the proposals everyone.

Given the updated requirements [here](https://github.com/Expensify/App/issues/42425#issuecomment-2148240788), @Krishna2323's main [solution](https://github.com/Expensify/App/issues/42425#issuecomment-2122885054) works good. Let's go with them.

🎀 👀 🎀 C+ reviewed

**Comment by allroundexperts** [comment]
Hi @Krishna2323!

I have messaged @JmillsExpensify. Thanks for your patience. Someone from the team should be handling this soon.

**Comment by allroundexperts** [comment]
## Checklist
1. https://github.com/Expensify/App/pull/30829
2. https://github.com/Expensify/App/pull/30829/files#r1713070836
3. N/A
4. A checklist would be helpful here.

## Regression test
1. Open any chat or room
2. Upload a video attachment
3. Play the video
4. Click on the "..." button
5. Click on the "Playback speed" button
6. Verify that the first option is highlighted
7. Change the speed
8. Click on the "..." button
9. Click on the "Playback speed" button again
10. Verify the selected speed is highlighted

Do we 👍 or 👎 ?

**Comment by JmillsExpensify** [comment]
@Krishna2323 All paid out. Always feel free to reach out to me via Slack if you don't see a response here.

**Comment by JmillsExpensify** [comment]
Payment summary: 

 - Contributor: @Krishna2323 $250 
 - Reviewer: @allroundexperts $250

**Comment by trjExpensify** [comment]
^^ Payment summary [here](https://github.com/Expensify/App/issues/42425#issuecomment-2284142563) confirmed is good.

**Comment by JmillsExpensify** [comment]
$250 approved for @allroundexperts
